### ISSUE TITLE
[When using yarn PNP] TypeError: Cannot read properties of undefined (reading 'on')

### ISSUE THREAD

**Issue by simPod** [vercel/next.js#46993]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 19.6.1
      npm: 9.4.0
      Yarn: 4.0.0-rc.40
      pnpm: N/A
    Relevant packages:
      next: 13.2.4
      eslint-config-next: 13.2.4
      react: 18.2.0
      react-dom: 18.2.0
```

### To Reproduce

Use yarn pnp and build app.

It was introduces in this PR https://github.com/vercel/next.js/pull/46705 I believe

https://github.com/vercel/next.js/blob/05f6de10869021dbb6124e174778e2108c9b4e8e/packages/next/src/lib/worker.ts#L51

### Describe the Bug

```
TypeError: Cannot read properties of undefined (reading 'on')
    at createWorker (/src/.yarn/__virtual__/next-virtual-9c7084ba23/4/build-cache/yarn/cache/next-npm-13.2.4-eedd00f0a6-9.zip/node_modules/next/dist/lib/worker.js:32:31)
    at new Worker (/src/.yarn/__virtual__/next-virtual-9c7084ba23/4/build-cache/yarn/cache/next-npm-13.2.4-eedd00f0a6-9.zip/node_modules/next/dist/lib/worker.js:42:9)
    at /src/.yarn/__virtual__/next-virtual-9c7084ba23/4/build-cache/yarn/cache/next-npm-13.2.4-eedd00f0a6-9.zip/node_modules/next/dist/build/index.js:593:35
    at async Span.traceAsyncFn (/src/.yarn/__virtual__/next-virtual-9c7084ba23/4/build-cache/yarn/cache/next-npm-13.2.4-eedd00f0a6-9.zip/node_modules/next/dist/trace/trace.js:79:20)
    at async Object.build [as default] (/src/.yarn/__virtual__/next-virtual-9c7084ba23/4/build-cache/yarn/cache/next-npm-13.2.4-eedd00f0a6-9.zip/node_modules/next/dist/build/index.js:73:2
```

### Expected Behavior

No error. 13.2.1 works

**Comment by timmywil** [comment]
I get the same error using regular old npm. Just running `next build`. No error in 13.2.3.

**Comment by timmywil** [comment]
Ok, after some digging, this can be reproduced with a fresh project created with `npx create-next-app@latest`, but only with the following in the next.config.js:

```js
experimental: {
  workerThreads: true
}
```

I forget why I added that. I actually used yarn pnp in the past and switched back to default npm so it's likely an artifact leftover from some solution I found in nextjs issues to fix a yarn pnp issue.

Removing that does seem to fix it, but I don't know if that will be possible for everyone.
